Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +15,214 @@ public static class BalanceStats
         /// </summary>
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
-        }
+            if (IsNullOrEmpty(peopleAndBalances)) return Constants.Nothing;
+            Thread.CurrentThread.CurrentCulture = Config.Culture();
 
+            var (names, money, isPlural) = ReturnNamesMoneyAndIsPlural(peopleAndBalances, "max");
+
+            return $"{names} had the most money ever. {FixMoneyFormat(money)}.";
+        }
+        
         /// <summary>
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
+            if (IsNullOrEmpty(peopleAndBalances)) return Constants.Nothing;
+            Thread.CurrentThread.CurrentCulture = Config.Culture();
+
+            var (names, money, isPlural) = ReturnNamesMoneyAndIsPlural(peopleAndBalances, "min");
+            if (money >= 0) return "N/A.";
+
+            return $"{names} lost the most money. {FixMoneyFormat(money)}.";
         }
 
         /// <summary>
         /// Return name and current money of the richest person.
         /// </summary>
         public static string FindRichestPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (IsNullOrEmpty(peopleAndBalances)) return Constants.Nothing;
+            Thread.CurrentThread.CurrentCulture = Config.Culture();
```

Comment:
I was expecting just Console.OutputEncoding to be changed. I think changing more stuff might have unexpected consequences (for example this now changes DateTime formatting as well). You just want to change text currency formatting and nothing else.

A+ for effort and originality though!